### PR TITLE
DG wk10 diagrams

### LINKED ISSUE #125: [Developer Guide] Adjust Model Component

### PR DESCRIPTION
**Pull request by shuenj**
Add model diagram in DG and AffiliationHistory explanation.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -136,10 +136,7 @@ The `Model` component,
 
 <box type="info" seamless>
 
-**Note:** An alternative (arguably, a more OOP) model is given below. It has a `Affiliation` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Affiliation` object per unique tag, instead of each `Person` needing their own `Affiliation` objects.<br>
-
-<puml src="diagrams/BetterModelClassDiagram.puml" width="450" />
-
+![Class diagram for Model](diagrams/modelPersonDiagram.png)
 </box>
```
Comment (leezhanpeng) [inline docs/DeveloperGuide.md]: Is it intended for the newly added Person class diagram be encapsulated in an alert indentation?

Also, maybe having a description explaining the newly added diagram will be useful here.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -171,99 +168,23 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+#### Implementation
+The `affiliationHistory` records the past and current affiliations that a person has with others. It is implemented similar to `affiliations`, but with a few differences:
+* When adding a person, the user specified `affiliations` will be added to `affiliationHistory` of himself/herself and the affiliated person.
+  * The `Person` constructor has been modified to use `affiliations` as the `affiliationHistory` during initialization.
+* When editing a `Person`'s `Name`, the user specified affiliation will also be updated in `affiliationHistory` of others.
+This is done by iterating through the affiliation list of the `Person` and updating the `affiliationHistory` of others. 
+* When editing a `Person`'s `affiliations`, specifically,
+  * adding `affiliations` will be add it to `affiliationHistory` as well
+  * removing an `affiliations`, will **not** be removed from `affiliationHistory`
+* When deleting a `Person`, the `affiliationHistory` of others will be updated accordingly.
```
Comment (leezhanpeng) [inline docs/DeveloperGuide.md:183]: Maybe this is an opportunity to add a diagram representing this interaction? Could be a sequence or an activity diagram.

File: docs/diagrams/modelPersonDiagram.png
Comment (leezhanpeng) [inline docs/diagrams/modelPersonDiagram.png:1]: Maybe making the triangle larger for the inheritance can provide better clarity for your diagram?

Adding multiplicity might provide a clearer picture as well.

Comment (Seonlo99) [inline docs/diagrams/modelPersonDiagram.png:1]: Is it possible to remove the red underline of the 3 words? Everything else looks good.

**Review by Seonlo99 — APPROVED** [review]
LGTM, nice work.

**Review by leezhanpeng — APPROVED** [review]
LGTM! Added some comments that you can consider if you are going to further adjust the DG parts later on.

File: docs/diagrams/modelPersonDiagram.png
Comment (leezhanpeng) [reply docs/diagrams/modelPersonDiagram.png:1]: Oh good catch, I missed it as well.

Comment (shuenj) [reply docs/diagrams/modelPersonDiagram.png:1]: Yep, done!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (shuenj) [reply docs/DeveloperGuide.md]: Ah, good catch, added description

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (shuenj) [reply docs/DeveloperGuide.md:183]: Added activity diagram!

File: docs/diagrams/modelPersonDiagram.png
Comment (shuenj) [reply docs/diagrams/modelPersonDiagram.png:1]: Enlarged the arrow, and added some multiplicities

**Pull request merged**